test/tests/TestUser.java

```diff
@@ -11,10 +15,44 @@ public TestUser(ComputationCoordinatorImp coordinator) {
   }
 
   public void run(String outputPath) throws Exception {
+    // Correct the path to match where your CSV file is located
+    String configPath = "test" + File.separatorChar + "tests" + File.separatorChar + "testInputFile.csv"; // Updated path
     char delimiter = ';';
-    String inputPath = "test" + File.separatorChar + "testInputFile.test";
 
-    // Call the appropriate method on the coordinator to run the computation
-    coordinator.beginComputation(delimiter, inputPath, delimiter, delimiter, delimiter, delimiter, inputPath, delimiter);
+    // Read configuration values from the CSV file
+    try (BufferedReader br = new BufferedReader(new FileReader(configPath))) {
+      br.readLine(); // Skip header row if there is one
+      String line;
+      if ((line = br.readLine()) != null) {
+        String[] config = line.split(",");
+
+        // Parse values from CSV line
+        int inputType = Integer.parseInt(config[0].trim());
+        String inputFile = config[1].trim();
+        int numGenerate = Integer.parseInt(config[2].trim());
+        int numRows = Integer.parseInt(config[3].trim());
+        int numCols = Integer.parseInt(config[4].trim());
+        int outputType = Integer.parseInt(config[5].trim());
+        String outputFileName = config[6].trim();
+        int outOrCompute = Integer.parseInt(config[7].trim());
+
+        // Print parsed values to console for verification
+        System.out.println("Parsed Configuration Values:");
+        System.out.println("Input Type: " + inputType);
+        System.out.println("Output File: " + inputFile);
+        System.out.println("Number to Generate: " + numGenerate);
+        System.out.println("Number of Rows: " + numRows);
+        System.out.println("Number of Columns: " + numCols);
+        System.out.println("Output Type: " + outputType);
+        System.out.println("Output File Name: " + outputFileName);
+        System.out.println("Out or Compute: " + outOrCompute);
+
+        // Call the computation method with parsed values
```

Comment:
i like that you added a file reader, all though i believe this should have been handled by DS

Reply:
Yeah i agree, we will implement the csv file config in the main code in a future pr.